tests/pc/test_retry_count.py

```diff
@@ -0,0 +1,361 @@
+            for pkt in lacpduPackets:
+                if pkt["LACPRetryCount"].version == 0xf1:
+                    if pkt["LACPRetryCount"].actor_system == actorInfo["system"] \
+                            and pkt["LACPRetryCount"].actor_port_number == actorInfo["port"] \
+                            and pkt["LACPRetryCount"].partner_system == partnerInfo["system"] \
+                            and pkt["LACPRetryCount"].partner_port_number == partnerInfo["port"]:
+                        sendVersionVerified = True
+                    elif pkt["LACPRetryCount"].actor_system == partnerInfo["system"] \
+                            and pkt["LACPRetryCount"].actor_port_number == partnerInfo["port"] \
+                            and pkt["LACPRetryCount"].partner_system == actorInfo["system"] \
+                            and pkt["LACPRetryCount"].partner_port_number == actorInfo["port"]:
+                        receiveVersionVerified = True
+            pytest_assert(sendVersionVerified, "unable to verify that LACPDU packets sent were the right version")
+            pytest_assert(receiveVersionVerified,
+                          "unable to verify that LACPDU packets received were the right version")
+
+
+@pytest.fixture(scope="function")
+def disable_retry_count_on_peer(request, nbrhosts, higher_retry_count_on_peers):
+    for nbr in list(nbrhosts.keys()):
+        nbrhosts[nbr]['host'].shell("teamdctl PortChannel1 state item set runner.enable_retry_count_feature false")
+
+    # Wait 90 seconds for retry count to get updated on the DUT
+    time.sleep(90)
+
+    yield
+
+    for nbr in list(nbrhosts.keys()):
+        nbrhosts[nbr]['host'].shell("teamdctl PortChannel1 state item set runner.enable_retry_count_feature true")
+
+
+@pytest.fixture(scope="function")
+def disable_retry_count_on_dut(request, duthost, higher_retry_count_on_dut):
+    cfg_facts = duthost.config_facts(host=duthost.hostname, source="running")["ansible_facts"]
+    for port_channel in list(cfg_facts["PORTCHANNEL"].keys()):
+        duthost.shell("teamdctl {} state item set runner.enable_retry_count_feature false".format(port_channel))
+
+    # Wait 90 seconds for retry count to get updated on the peers
+    time.sleep(90)
+
+    yield
+
+    for port_channel in list(cfg_facts["PORTCHANNEL"].keys()):
+        duthost.shell("teamdctl {} state item set runner.enable_retry_count_feature true".format(port_channel))
+
+
+class TestNeighborRetryCount:
+    def test_peer_retry_count(self, duthost, nbrhosts, higher_retry_count_on_peers):
+        """
+        Test that DUT sees new retry count when peers update retry count.
+        """
+        for nbr in list(nbrhosts.keys()):
+            port_channel_status = nbrhosts[nbr]['host'].get_port_channel_status("PortChannel1")
+            pytest_assert(port_channel_status["runner"]["retry_count"] == 5, "retry count on neighbor is incorrect")
+            pytest_assert(port_channel_status["runner"]["retry_count"] == 5,
```

Comment:
This does not seem to be checking DUT's retry count. The check for both DUT and peer are same, so something is off here.

Reply:
This might be a combination of copy-paste errors, I'll fix it.